connectors/src/connectors/notion/temporal/workflows/index.ts

```diff
@@ -258,6 +259,14 @@ export async function notionGarbageCollectionWorkflow({
         forceResync: false,
       })
     );
+
+    if (pageIndex > 512) {
+      await continueAsNew<typeof notionGarbageCollectionWorkflow>({
```

Comment:
as per IRL, we'll need to await all created promises right before this line otherwise we're losing a lot of the work that has been enqueued and we won't process everything.

Additionally, we may also want to pass through the `pageIndex` in workflow params and do pageIndex % 512 === 0 so we keep the info in the workflow IDs which is often useful when inspecting temporal UI 👍

Reply:
Thanks :pray: